library/core/src/num/dec2flt/algorithm.rs

```diff
@@ -78,7 +84,13 @@ mod fpu_precision {
         // `FPUControlWord` structure is dropped
         // SAFETY: the `fnstcw` instruction has been audited to be able to work correctly with
         // any `u16`
-        unsafe { llvm_asm!("fnstcw $0" : "=*m" (&cw) ::: "volatile") }
+        unsafe {
+            asm!(
+                "fnstcw [{:e}]",
+                lateout(reg) cw,
```

Comment:
```suggestion
                in(reg) &mut cw,
```

Reply:
My bad. So with instructions write to memory address operands, we use `in(reg) &mut <var>`.

Reply:
Correct, you want to end up with `fnstcw [eax]` as the assembly string where `eax` contains the address of `cw`.

Reply:
Thanks for confirming.